experimental/incremental/task.go

```diff
@@ -43,6 +58,13 @@ type Task struct {
 
 	// Intrusive linked list node for cycle detection.
 	path path
+
+	// Set if we're currently holding the executor's semaphore. This exists to
+	// ensure that we do not violate concurrency assumptions, and is never
+	// itself mutated concurrently.
+	holding bool
+	// True if this task is executing on the g that called [Run].
+	rootg bool
```

Comment:
Maybe just `root` instead
```suggestion
	root bool
```